ManagedIdentityWebApi/Controllers/AppServiceController.cs

```diff
@@ -0,0 +1,108 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Identity.Client;
+using Microsoft.IdentityModel.Abstractions;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace ManagedIdentityWebApi.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    [SwaggerTag(description: "Get MSI Token")]
+    public class AppServiceController : ControllerBase
+    {
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Default constructor
+        /// </summary>
+        /// <param name="logger"></param>
+        public AppServiceController(ILogger<AppServiceController> logger)
+        {
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Acquires token for managed identity
+        /// </summary>
+        /// <param name="resourceUri"></param>
+        /// <param name="userAssignedId"></param>
+        /// <returns></returns>
+        [HttpGet]
+        [SwaggerResponse(200, "Returns an Azure Resource MSI Token Response", Type = typeof(HttpResponse))]
+        [SwaggerResponse(400, "Returns the error object for any validation failures", Type = typeof(HttpResponse))]
+        [SwaggerResponse(500, "Returns the error object for any Server Errors", Type = typeof(HttpResponse))]
+        public async Task<string> GetAsync([FromQuery(Name = "resourceuri")] string? resourceUri,
+            [FromQuery(Name = "userAssignedId")] string? userAssignedId = null)
+        {
+            _logger.LogInformation("Get token from MSAL for managed identity.");
+            try
+            {
+                IManagedIdentityApplication mi = CreateManagedIdentityApplication(userAssignedId);
+
+                var result = await mi.AcquireTokenForManagedIdentity(resourceUri).ExecuteAsync().ConfigureAwait(false);
+
+                return "Access token received. Token Source: " + result.AuthenticationResultMetadata.TokenSource;
+            }
+            catch (MsalException ex)
```

Comment:
```suggestion
            catch (MsalException ex)
```
thought we had a managedidentityexception? no?

Reply:
No we need to add one

Reply:
Ok if we do not have a tracking item for the items we agreed with Azure SDK, we should create those